In [33]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)

file_path = "C:/Users/bapti/OneDrive/Documentos/Insper/Ps jr/hackathon-ambev-dados/data/raw/Analise_LongNeck_WSNP - Sem repostas.xlsb"

xls = pd.ExcelFile(file_path)
xls.sheet_names

['Cenário atual BR',
 'Custos de transferência',
 'Produção PCP',
 'Transferências Programadas',
 'Cenário Divulgado',
 'Cenário com Nova Demanda']

In [34]:
df = pd.read_excel(xls, sheet_name="Cenário atual BR", header=None, engine="pyxlsb")

df_data = df.iloc[:13, :]
df_data

,0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23
0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,NaN,Janeiro,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Fevereiro,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,NaN,NaN,Janeiro,26,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Fevereiro,24,NaN,NaN,NaN,NaN,NaN,NaN
3,NaN,NaN,Demanda,NaN,NaN,PROD,NaN,NaN,NaN,NaN,Estoque,NaN,NaN,NaN,NaN,NaN,Demanda,NaN,NaN,PROD,NaN,Estoque,NaN,NaN
4,NaN,SK 269,GEO/REG,Demanda,NaN,Real,WSNP,Real,1W,NaN,EI Mês,Suf.ini(d),Transf. Malha,EFM,Suf. f (d),NaN,GEO/REG,Demanda,NaN,WSNP,NaN,Transf. Malha,EFM,Suf. f (d)
5,NaN,LONG NECK,MG,65538.324,NaN,0,0,0,0,NaN,28580.4,11.338258,11585.87334,32771.46546,10.796472,NaN,MG,72849.276,NaN,0,NaN,75118.432397,35040.621857,12.893728
6,NaN,NaN,SP,173637.414,NaN,127269,15300,256624.2,238579.2,NaN,114472.8,17.14085,-12423.928037,120086.451823,17.439383,NaN,SP,165262.428,NaN,253602.257143,NaN,-75583.096226,132843.18474,25.529185
7,NaN,NaN,NENO,200754.27,NaN,70399,9342.857143,143535.342857,176297.4,NaN,136190,15.388772,7455.631731,86426.704589,11.544483,NaN,NENO,179673.786,NaN,158000,NaN,34507,99259.918589,13.258685
8,NaN,NaN,CO,104493.312,NaN,50470,5142.857143,100103.142857,99590.4,NaN,61952.4,15.414981,7619.571411,81961.587309,17.438284,NaN,CO,112802.274,NaN,87685.714286,NaN,21793.962266,78638.98986,21.104261
9,NaN,NaN,RJ,111219.408,NaN,130770,17571.428571,267014.571429,339789.6,NaN,84834,19.831826,-29886.918917,87765.186831,16.122267,NaN,RJ,130649.4,NaN,288282.857143,NaN,-174400.90668,70997.737294,20.304152


In [35]:
df_jan = df.iloc[5:13, [2,3,7,8,10,11,12,13,14]].copy()
df_jan = df_jan[df_jan[2] != "Export"]
df_jan = df_jan.reset_index(drop=True)

df_jan


,2,3,7,8,10,11,12,13,14
0,MG,65538.324,0,0,28580.4,11.338258,11585.87334,32771.46546,10.796472
1,SP,173637.414,256624.2,238579.2,114472.8,17.14085,-12423.928037,120086.451823,17.439383
2,NENO,200754.27,143535.342857,176297.4,136190,15.388772,7455.631731,86426.704589,11.544483
3,CO,104493.312,100103.142857,99590.4,61952.4,15.414981,7619.571411,81961.587309,17.438284
4,RJ,111219.408,267014.571429,339789.6,84834,19.831826,-29886.918917,87765.186831,16.122267
5,SUL,116261.406,81343.285714,104891.4,86758.2,19.402081,12433.157717,77470.006191,15.947827
6,TOTAL,838959.534,848620.542857,959148,495419.4,15.353428,-4.486706,486481.402203,13.525908


In [36]:
df_fev = df.iloc[5:13, [16,17,19,21,22,23]].copy()
df_fev = df_fev[df_fev[16] != "Export"]
df_fev = df_fev.reset_index(drop=True)

df_fev


,16,17,19,21,22,23
0,MG,72849.276,0,75118.432397,35040.621857,12.893728
1,SP,165262.428,253602.257143,-75583.096226,132843.18474,25.529185
2,NENO,179673.786,158000,34507,99259.918589,13.258685
3,CO,112802.274,87685.714286,21793.962266,78638.98986,21.104261
4,RJ,130649.4,288282.857143,-174400.90668,70997.737294,20.304152
5,SUL,116585.172,77748.428571,38221.732389,76854.995151,17.44864
6,TOTAL,827001.936,863199.257143,-118.019829,493635.447491,14.325542


In [37]:
df_jan.columns = [
    "GEO/REG",
    "Demanda",
    "Prod_Real",
    "Prod_1W",
    "EI_Mes",
    "Suf_ini_d",
    "Transf_Malha",
    "EFM",
    "Suf_f_d"
]

df_fev.columns = [
    "GEO/REG",
    "Demanda",
    "WSNP",
    "Transf_Malha",
    "EFM",
    "Suf_f_d"
]

In [38]:
for col in df_jan.columns[1:]:
    df_jan[col] = pd.to_numeric(df_jan[col], errors="coerce")
df_jan.dtypes

for col in df_fev.columns[1:]:
    df_fev[col] = pd.to_numeric(df_fev[col], errors="coerce")
df_fev.dtypes

GEO/REG             str
Demanda         float64
WSNP            float64
Transf_Malha    float64
EFM             float64
Suf_f_d         float64
dtype: object

In [39]:
df_jan = df_jan.round()
df_jan

,GEO/REG,Demanda,Prod_Real,Prod_1W,EI_Mes,Suf_ini_d,Transf_Malha,EFM,Suf_f_d
0,MG,65538.0,0.0,0.0,28580.0,11.0,11586.0,32771.0,11.0
1,SP,173637.0,256624.0,238579.0,114473.0,17.0,-12424.0,120086.0,17.0
2,NENO,200754.0,143535.0,176297.0,136190.0,15.0,7456.0,86427.0,12.0
3,CO,104493.0,100103.0,99590.0,61952.0,15.0,7620.0,81962.0,17.0
4,RJ,111219.0,267015.0,339790.0,84834.0,20.0,-29887.0,87765.0,16.0
5,SUL,116261.0,81343.0,104891.0,86758.0,19.0,12433.0,77470.0,16.0
6,TOTAL,838960.0,848621.0,959148.0,495419.0,15.0,-4.0,486481.0,14.0


In [40]:
df_fev = df_fev.round()
df_fev

,GEO/REG,Demanda,WSNP,Transf_Malha,EFM,Suf_f_d
0,MG,72849.0,0.0,75118.0,35041.0,13.0
1,SP,165262.0,253602.0,-75583.0,132843.0,26.0
2,NENO,179674.0,158000.0,34507.0,99260.0,13.0
3,CO,112802.0,87686.0,21794.0,78639.0,21.0
4,RJ,130649.0,288283.0,-174401.0,70998.0,20.0
5,SUL,116585.0,77748.0,38222.0,76855.0,17.0
6,TOTAL,827002.0,863199.0,-118.0,493635.0,14.0


In [41]:
df_jan.columns = pd.MultiIndex.from_product(
    [["Janeiro"], df_jan.columns]
)

df_fev.columns = pd.MultiIndex.from_product(
    [["Fevereiro"], df_fev.columns]
)

df_final = pd.concat([df_jan, df_fev], axis=1)
df_final

Janeiro                                                                 \
  GEO/REG   Demanda Prod_Real   Prod_1W    EI_Mes Suf_ini_d Transf_Malha   
0      MG   65538.0       0.0       0.0   28580.0      11.0      11586.0   
1      SP  173637.0  256624.0  238579.0  114473.0      17.0     -12424.0   
2    NENO  200754.0  143535.0  176297.0  136190.0      15.0       7456.0   
3      CO  104493.0  100103.0   99590.0   61952.0      15.0       7620.0   
4      RJ  111219.0  267015.0  339790.0   84834.0      20.0     -29887.0   
5     SUL  116261.0   81343.0  104891.0   86758.0      19.0      12433.0   
6   TOTAL  838960.0  848621.0  959148.0  495419.0      15.0         -4.0   

                    Fevereiro                                             \
        EFM Suf_f_d   GEO/REG   Demanda      WSNP Transf_Malha       EFM   
0   32771.0    11.0        MG   72849.0       0.0      75118.0   35041.0   
1  120086.0    17.0        SP  165262.0  253602.0     -75583.0  132843.0   
2   86427.0    12.0      NENO  179674.0  158000.0      34507.0   99260.0   
3   81962.0    17.0        CO  112802.0   87686.0      21794.0   78639.0   
4   87765.0    16.0        RJ  130649.0  288283.0    -174401.0   70998.0   
5   77470.0    16.0       SUL  116585.0   77748.0      38222.0   76855.0   
6  486481.0    14.0     TOTAL  827002.0  863199.0       -118.0  493635.0   

           
  Suf_f_d  
0    13.0  
1    26.0  
2    13.0  
3    21.0  
4    20.0  
5    17.0  
6    14.0

In [70]:
df_compare = pd.DataFrame({
    "Jan": df_jan[("Janeiro", "Demanda")].values,
    "Fev": df_fev[("Fevereiro", "Demanda")].values
}, index=df_jan[("Janeiro", "GEO/REG")])

df_compare["Var_abs"] = df_compare["Fev"] - df_compare["Jan"]
df_compare["Var_%"] = (df_compare["Var_abs"] / df_compare["Jan"]) * 100

df_compare.index.name = "GEO/REG"
df_compare = df_compare.reset_index()
df_compare = df_compare.round(2)

def cor_variacao(val):
    if val > 0:
        return "color: green"
    elif val < 0:
        return "color: red"
    return ""

df_compare_cor = df_compare.style \
    .format({
        "Jan": "{:,.0f}",
        "Fev": "{:,.0f}",
        "Var_abs": "{:,.0f}",
        "Var_%": "{:.2f}%"
    }) \
    .map(cor_variacao, subset=["Var_abs", "Var_%"])

df_compare_cor

,GEO/REG,Jan,Fev,Var_abs,Var_%
0,MG,"65,538","72,849","7,311",11.16%
1,SP,"173,637","165,262","-8,375",-4.82%
2,NENO,"200,754","179,674","-21,080",-10.50%
3,CO,"104,493","112,802","8,309",7.95%
4,RJ,"111,219","130,649","19,430",17.47%
5,SUL,"116,261","116,585",324,0.28%
6,TOTAL,"838,960","827,002","-11,958",-1.43%


Análise da demanda - Janeiro vs Fevereiro

Observa-se que a demanda total apresentou uma queda de 11.958 unidades no período, indicando leve desaceleração no volume consolidado entre Janeiro e Fevereiro.

A variação entretanto não foi homogênea entre as regiões. O crescimento expressivo no RJ (+19,430) e os aumentos registrados em MG e CO não foram suficientes para compensar a queda acentudada em NENO (-21.080) e a redução em SP (-8.355).

O cenário indica um mercado heterogêneo, com crescimento concentrado em algumas regiões e retração relevante em outras. A principal conclusão é que a queda consolidada foi puxada por regiões específicas (especialmente NENO), e não por um enfraquecimento uniforme da demanda.



